### PR TITLE
Added HeaderLocator to identify tenants from header information

### PR DESCRIPTION
**Pull request by kieljohn**

### CODE REVIEW

File: src/Vivait/TenantBundle/Locator/HeaderLocator.php [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        $this->request = $request;
+        $this->tenant_key = $tenant_key;
+    }
+
+    /**
+     * @return string Tenant key
+     */
+    public function getTenant()
+    {
+        if (!$this->request->headers->has($this->tenant_key)) {
```
Comment (jackbentley) [inline src/Vivait/TenantBundle/Locator/HeaderLocator.php]: `tenant_key` is a bit confusing. Perhaps change it to something along the lines of `headerField`?

File: src/Vivait/TenantBundle/Locator/HeaderLocator.php [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+            throw new \RuntimeException('Could not locate a tenant in the header');
```
Comment (jackbentley) [inline src/Vivait/TenantBundle/Locator/HeaderLocator.php]: What kind of feedback does this give back to the end user? I presume a 500 error page?

Comment (LewisW) [inline src/Vivait/TenantBundle/Locator/HeaderLocator.php]: Symfony should have registered its exception handler by this point, but I'll double check. Although if it hasn't, really it's a problem for the tenant bundle, not this driver.

**Comment by LewisW** [comment]
Nice work. Ignore the tests failing, it's an unrelated test that has always failed for the OS that travis uses.

**Pull request merged**